### PR TITLE
MDEV-14871 Server crashes in fill_record / fill_record_n_invoke_before_triggers upon inserting into versioned table with trigger

### PR DESCRIPTION
**Pull request by kevgs**
```c++
#0  Field::make_new_field (this=0x7fffdc006c30, root=0x7fffdc18eea0, new_table=0x7fffdc18e3e0, keep_type=true) at field.cc:2243
#1  Table_triggers_list::prepare_record_accessors (this=0x7fffdc1226c8, table=0x7fffdc18e3e0) at sql_trigger.cc:1234
#2  Table_triggers_list::check_n_load (thd=0x7fffdc000d50, db=0x7fffdc0aae38 "test", table_name=0x7fffdc0aae3d "t1", table=0x7fffdc18e3e0, names_only=false) at sql_trigger.cc:1593
#3  open_table_entry_fini (thd=0x7fffdc000d50, share=0x7fffdc0aa7c8, entry=0x7fffdc18e3e0) at sql_base.cc:2682
#4  open_table (thd=0x7fffdc000d50, table_list=0x7fffdc015030, ot_ctx=0x7fffeebc5ba8) at sql_base.cc:1926
#5  open_and_process_table (thd=0x7fffdc000d50, lex=0x7fffdc004a20, tables=0x7fffdc015030, counter=0x7fffeebc5cac, flags=0, prelocking_strategy=0x7fffeebc5d20, has_prelocking_list=false, ot_ctx=0x7fffeebc5ba8) at sql_base.cc:3467
#6  open_tables (thd=0x7fffdc000d50, options=..., start=0x7fffeebc5cc0, counter=0x7fffeebc5cac, flags=0, prelocking_strategy=0x7fffeebc5d20) at sql_base.cc:3985
#7  open_and_lock_tables (thd=0x7fffdc000d50, options=..., tables=0x7fffdc015030, derived=true, flags=0, prelocking_strategy=0x7fffeebc5d20) at sql_base.cc:4742
#8  open_and_lock_tables (thd=0x7fffdc000d50, tables=0x7fffdc015030, derived=true, flags=0) at sql_base.h:494
#9  mysql_insert (thd=0x7fffdc000d50, table_list=0x7fffdc015030, fields=List<Item> with 0 elements, values_list=..., update_fields=List<Item> with 0 elements, update_values=List<Item> with 0 elements, duplic=DUP_ERROR, ignore=false) at sql_insert.cc:766
#10 mysql_execute_command (thd=0x7fffdc000d50) at sql_parse.cc:4704
#11 mysql_parse (thd=0x7fffdc000d50, rawbuf=0x7fffdc014f48 "insert into t1 values (1),(2)", length=29, parser_state=0x7fffeebc9410, is_com_multi=false, is_next_command=false) at sql_parse.cc:7991
```

### CODE REVIEW

File: sql/field.cc [hunk 1]
```diff
@@ -2240,7 +2240,6 @@ Field *Field::make_new_field(MEM_ROOT *root, TABLE *new_table,
                 VERS_SYS_START_FLAG | VERS_SYS_END_FLAG |
                 VERS_UPDATE_UNVERSIONED_FLAG);
   tmp->reset_fields();
-  tmp->invisible= VISIBLE;
```
Comment (midenok) [inline sql/field.cc]: Did you test with `main.invisible_binlog main.invisible_field_debug main.invisible_field`?

File: mysql-test/suite/versioning/t/insert.test [hunk 2]
```diff
@@ -251,4 +251,12 @@ create table t1 (a int) with system versioning;
 insert t1 values (1);
 select * from t1 for system_time as of now(6);
 select * from t1 for system_time as of sysdate(6);
+
+#
+# MDEV-14871 Server crashes in fill_record / fill_record_n_invoke_before_triggers upon inserting into versioned table with trigger
+#
```
Comment (midenok) [inline mysql-test/suite/versioning/t/insert.test]: Use `--echo` in comments to put them into result.

File: sql/field.cc (hunk 1 above)
Comment (kevgs) [reply sql/field.cc]: That was incorrect fix. I've read `git blame` incorrectly. Thanks for noticing.

File: mysql-test/suite/versioning/t/insert.test (hunk 2 above)
Comment (kevgs) [reply mysql-test/suite/versioning/t/insert.test]: Fixed.

**Review by midenok — APPROVED** [review]

**Pull request merged**